src/io/dataset_loader.cpp

```diff
@@ -933,8 +948,22 @@ void DatasetLoader::ConstructBinMappersFromTextData(int rank, int num_machines,
   const data_size_t filter_cnt = static_cast<data_size_t>(
     static_cast<double>(config_.min_data_in_leaf* sample_data.size()) / dataset->num_data_);
 
+  bool force_findbin_in_single_machine = false;
+  if (Network::num_machines() > 1) {
+    int total_num_feature = Network::GlobalSyncUpByMin(dataset->num_total_features_);
+    size_t esimate_sync_size = BinMapper::SizeForSpecificBin(config_.max_bin) * total_num_feature;
```

Comment:
Same as above, avoids overflow.
```suggestion
    size_t esimate_sync_size = (size_t)BinMapper::SizeForSpecificBin(config_.max_bin) * (size_t)total_num_feature;
```